tests/unit/remote/test_git.py

```diff
@@ -39,6 +40,42 @@ def test_is_not_repo(new_dir):
     assert not is_repo(new_dir)
 
 
+def test_is_shallow_repo(new_dir):
```

Comment:
Is there a test for calling this function on a repo that is not shallow?
What about on a path that is not a repo?

Reply:
The function call `is_repo` first, unless you want a nearly duplicate tests to do it.

Reply:
Yes, I do want to see this explicitly tested. We should test all branches in a function when feasible.

Reply:
OK, added